blueprints/data-solutions/vertex-mlops/main.tf

```diff
@@ -0,0 +1,230 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+locals {
+  service_encryption_keys = var.service_encryption_keys
+  shared_vpc_project      = try(var.network_config.host_project, null)
+
+  subnet = (
+    local.use_shared_vpc
+    ? var.network_config.subnet_self_link
+    : values(module.vpc-local.0.subnet_self_links)[0]
+  )
+  vpc = (
+    local.use_shared_vpc
+    ? var.network_config.network_self_link
+    : module.vpc-local.0.self_link
+  )
+  use_shared_vpc = var.network_config != null
+
+  shared_vpc_bindings = {
+    "roles/compute.networkUser" = [
+      "robot-df", "notebooks"
+    ]
+  }
+
+  shared_vpc_role_members = {
+    robot-df  = "serviceAccount:${module.project.service_accounts.robots.dataflow}"
+    notebooks = "serviceAccount:${module.project.service_accounts.robots.notebooks}"
+  }
+
+  # reassemble in a format suitable for for_each
+  shared_vpc_bindings_map = {
+    for binding in flatten([
+      for role, members in local.shared_vpc_bindings : [
+        for member in members : { role = role, member = member }
+      ]
+    ]) : "${binding.role}-${binding.member}" => binding
+  }
+}
+
+module "gcs-bucket" {
+  count          = var.bucket_name == null ? 0 : 1
+  source         = "../../../modules/gcs"
+  project_id     = module.project.project_id
+  name           = var.bucket_name
+  prefix         = var.prefix
+  location       = var.region
+  storage_class  = "REGIONAL"
+  versioning     = false
+  encryption_key = try(local.service_encryption_keys.storage, null) # Example assignment of an encryption key
+}
+
+# Default bucket for Cloud Build to prevent error: "'us' violates constraint ‘constraints/gcp.resourceLocations’"
+# https://stackoverflow.com/questions/53206667/cloud-build-fails-with-resource-location-constraint
+module "gcs-bucket-cloudbuild" {
+  source         = "../../../modules/gcs"
+  project_id     = module.project.project_id
+  name           = "${var.project_id}_cloudbuild"
+  prefix         = var.prefix
+  location       = var.region
+  storage_class  = "REGIONAL"
+  versioning     = false
+  encryption_key = try(local.service_encryption_keys.storage, null) # Example assignment of an encryption key
+}
+
+module "bq-dataset" {
+  count          = var.dataset_name == null ? 0 : 1
+  source         = "../../../modules/bigquery-dataset"
+  project_id     = module.project.project_id
+  id             = var.dataset_name
+  location       = var.region
+  encryption_key = try(local.service_encryption_keys.bq, null) # Example assignment of an encryption key
+}
+
+module "vpc-local" {
+  count      = local.use_shared_vpc ? 0 : 1
+  source     = "../../../modules/net-vpc"
+  project_id = module.project.project_id
+  name       = "default"
```

Comment:
I would not use `default` as name of the VPC. I would use one related to the blueprint.